src/charm.py

```diff
@@ -120,15 +142,33 @@ def _login_ui_layer(self) -> Layer:
         return Layer(pebble_layer)
 
     def _get_kratos_endpoint_info(self) -> Optional[str]:
-        kratos_public_url = ""
         if self.model.relations[self._kratos_relation_name]:
             try:
                 kratos_endpoints = self.kratos_endpoints.get_kratos_endpoints()
                 kratos_public_url = kratos_endpoints["public_endpoint"]
+                return kratos_public_url
             except KratosEndpointsRelationDataMissingError:
                 logger.info("No kratos-endpoint-info relation data found")
+                return None
```

Comment:
I think we shouldn't set `None` as environment variable values, I would opt to set empty strings instead

Reply:
fixed